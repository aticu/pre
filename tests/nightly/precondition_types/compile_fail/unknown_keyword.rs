use pre::pre;

#[pre(unknown_keyword)]
fn foo(unknown_keyword: bool) {}

#[pre]
fn main() {
    #[assure(unknown_keyword, reason = "`unknown_keyword` is `true`")]
    foo(true);
}
