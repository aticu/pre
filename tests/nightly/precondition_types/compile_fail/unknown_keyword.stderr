error: keyword `unknown_keyword` is not recognized by pre
 --> $DIR/unknown_keyword.rs:3:7
  |
3 | #[pre(unknown_keyword)]
  |       ^^^^^^^^^^^^^^^
  |
  = help: if you wanted to use a boolean expression, try `unknown_keyword == true`
